python_modules/dagster/dagster/__init__.py

```diff
@@ -244,11 +246,54 @@
     execute_solid_within_pipeline,
     execute_solids_within_pipeline,
 )
+from pep562 import pep562
 
 from .version import __version__
 
 from dagster.config.source import BoolSource, StringSource, IntSource  # isort:skip
 
+_DEPRECATED = {
+    "EventMetadataEntry": ("MetadataEntry", MetadataEntry),
+    "EventMetadata": ("MetadataValue", MetadataValue),
+    "TextMetadataEntryData": ("TextMetadataValue", TextMetadataValue),
+    "UrlMetadataEntryData": ("UrlMetadataValue", UrlMetadataValue),
+    "PathMetadataEntryData": ("PathMetadataValue", PathMetadataValue),
+    "JsonMetadataEntryData": ("JsonMetadataValue", JsonMetadataValue),
+    "MarkdownMetadataEntryData": ("MarkdownMetadataValue", MarkdownMetadataValue),
+    "PythonArtifactMetadataEntryData": ("PythonArtifactMetadataValue", PythonArtifactMetadataValue),
+    "FloatMetadataEntryData": ("FloatMetadataValue", FloatMetadataValue),
+    "IntMetadataEntryData": ("IntMetadataValue", IntMetadataValue),
+    "DagsterPipelineRunMetadataEntryData": (
+        "DagsterPipelineRunMetadataValue",
+        DagsterPipelineRunMetadataValue,
+    ),
+    "DagsterAssetMetadataEntryData": ("DagsterAssetMetadataValue", DagsterAssetMetadataValue),
+    "TableMetadataEntryData": ("TableMetadataValue", TableMetadataValue),
+    "TableSchemaMetadataEntryData": ("TableSchemaMetadataValue", TableSchemaMetadataValue),
+}
+
+
+def __getattr__(name):
+    if name in _DEPRECATED:
+        new_name, value = _DEPRECATED[name]
+        stacklevel = 3 if sys.version_info >= (3, 7) else 4
+        rename_warning(new_name, name, "0.15.0", stacklevel=stacklevel)
```

Comment:
seems bold to assume everything in `_DEPRECATED` will expire in `0.15.0` , assuming we are going to leverage this set-up more moving forward

Reply:
Excellent point, fixed